panic: failed to delete stream: failed to delete stream data directory

Hi!
Sometime LB doesn't start with that error:

```
/bin/liftbridge --config=/etc/liftbridge/liftbridge.yml
INFO [..12:46:04] Liftbridge Version:        v1.3.0            
INFO [..12:46:04] Server ID:                 centos-noc1       
INFO [..12:46:04] Namespace:                 liftbridge-default 
INFO [..12:46:04] Default Retention Policy:  [Age: 1 day, Compact: false] 
INFO [..12:46:04] Default Partition Pausing: disabled          
INFO [..12:46:04] Starting server on 172.22.2.229:9292...      
INFO [..12:46:05]  raft: initial configuration: index=1 servers="[{Suffrage:Voter ID:centos-noc1 Address:centos-noc1}]" 
DEBU[..12:46:05] Loaded existing state for metadata Raft group 
INFO[..12:46:05]  raft: entering follower state: follower="Node at centos-noc1 [Follower]" leader= 
DEBU[..12:46:05] api: FetchMetadata []                        
WARN[..12:46:06]  raft: heartbeat timeout reached, starting election: last-leader= 
INFO[..12:46:06]  raft: entering candidate state: node="Node at centos-noc1 [Candidate]" term=3558 
DEBU[..12:46:06] raft: votes: needed=1                        
DEBU[..12:46:06] raft: vote granted: from=centos-noc1 term=3558 tally=1 
INFO[..12:46:06]  raft: election won: tally=1                 
INFO[..12:46:06]  raft: entering leader state: leader="Node at centos-noc1 [Leader]" 
INFO[..12:46:06] Server became metadata leader, performing leader promotion actions 
DEBU[..12:46:07] fsm: Replaying Raft log...                   
panic: failed to delete stream: failed to delete stream data directory: remove /var/lib/liftbridge/streams/events.default: directory not empty
 
goroutine 39 [running]:
github.com/liftbridge-io/liftbridge/server.(*Server).Apply(0xc00037c000, 0xc00007c8c0, 0x0, 0x0)
        /home/circleci/project/server/fsm.go:111 +0x3e3
github.com/hashicorp/raft.(*Raft).runFSM.func1(0xc0000781d0)
        /go/pkg/mod/github.com/hashicorp/raft@v1.1.2/fsm.go:90 +0x2c1
github.com/hashicorp/raft.(*Raft).runFSM.func2(0xc000030200, 0x40, 0x40)
        /go/pkg/mod/github.com/hashicorp/raft@v1.1.2/fsm.go:113 +0x75
github.com/hashicorp/raft.(*Raft).runFSM(0xc000488000)
        /go/pkg/mod/github.com/hashicorp/raft@v1.1.2/fsm.go:219 +0x42f
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc000488000, 0xc000079ea0)
        /go/pkg/mod/github.com/hashicorp/raft@v1.1.2/state.go:146 +0x55
created by github.com/hashicorp/raft.(*raftState).goFunc
        /go/pkg/mod/github.com/hashicorp/raft@v1.1.2/state.go:144 +0x6
```
Directory indeed not empty, but what is the problem? 

It fixes only when I remove all data_dir
Also i can attach an archieve with LB data, if it needed.

--- a/server/api_test.go
+++ b/server/api_test.go
@@ -3,6 +3,8 @@ package server
 import (
 	"context"
 	"fmt"
+	"os"
+	"path/filepath"
 	"strconv"
 	"testing"
 	"time"
@@ -236,11 +238,17 @@ func TestDeleteStream(t *testing.T) {
 	err = client.CreateStream(context.Background(), "foo", "foo", lift.Partitions(3))
 	require.NoError(t, err)
 
+	_, err = os.Stat(filepath.Join(s1Config.DataDir, "streams", "foo"))
+	require.NoError(t, err)
+
 	stream := s1.metadata.GetStream("foo")
 	require.NotNil(t, stream)
 
 	err = client.DeleteStream(context.Background(), "foo")
 	require.NoError(t, err)
+
+	_, err = os.Stat(filepath.Join(s1Config.DataDir, "streams", "foo"))
+	require.True(t, os.IsNotExist(err))
 
 	stream = s1.metadata.GetStream("foo")
 	require.Nil(t, stream)

--- a/server/metadata.go
+++ b/server/metadata.go
@@ -835,7 +835,7 @@ func (m *metadataAPI) CloseAndDeleteStream(stream *stream) error {
 		return errors.Wrap(err, "failed to delete stream")
 	}
 
-	// Remove the (now empty) stream data directory
+	// Remove the stream data directory
 	streamDataDir := filepath.Join(m.Server.config.DataDir, "streams", stream.GetName())
 	err = os.RemoveAll(streamDataDir)
 	if err != nil {

--- a/server/metadata.go
+++ b/server/metadata.go
@@ -837,7 +837,7 @@ func (m *metadataAPI) CloseAndDeleteStream(stream *stream) error {
 
 	// Remove the (now empty) stream data directory
 	streamDataDir := filepath.Join(m.Server.config.DataDir, "streams", stream.GetName())
-	err = os.Remove(streamDataDir)
+	err = os.RemoveAll(streamDataDir)
 	if err != nil {
 		return errors.Wrap(err, "failed to delete stream data directory")
 	}

